src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -3636,3 +3636,98 @@ mcd_rpc_message_decompress_if_necessary (mcd_rpc_message *rpc, void **data, size
 
    return mcd_rpc_message_decompress (rpc, data, data_len);
 }
+
+bool
+mongoc_cluster_run_retryable_write (mongoc_cluster_t *cluster,
+                                    mongoc_cmd_t *cmd,
+                                    bool is_retryable_write,
+                                    mongoc_server_stream_t **retry_server_stream,
+                                    bson_t *reply,
+                                    bson_error_t *error)
+{
+   BSON_ASSERT_PARAM (cluster);
+   BSON_ASSERT_PARAM (cmd);
+   BSON_ASSERT_PARAM (retry_server_stream);
+   BSON_ASSERT_PARAM (reply);
+
+   bool ret;
+   // `can_retry` is set to false on retry. A retry may only happen once.
+   bool can_retry = is_retryable_write;
+
+   /* increment the transaction number for the first attempt of each retryable
+    * write command */
```

Comment:
```suggestion
   // Increment the transaction number for the first attempt of each retryable
   // write command.
```
Mix of `/**/` and `//` style comments is distracting. Recommend using `//` consistently.